### PR TITLE
Add agent workflow guide, PR template, and issue templates

### PR DESCRIPTION
**Pull request by aurablacklight**
## Summary

- **AGENTS.md**: Added a 7-step operational workflow for AI agents (branching, context reading, issue analysis, implementation, testing, PR creation, stuck protocol) while keeping existing project reference content
- **PR template**: Structured format with Summary, Changes, Testing checklist, Clean Room compliance checkbox, and issue link
- **Issue templates**: Bug report and feature request templates with acceptance criteria sections so agents know when to stop

## Motivation

We're running multiple AI agents concurrently on this repo. The existing AGENTS.md told agents *what the project is* but not *how to operate*. This caused inconsistent branching, missing tests, vague PRs, and agents spinning when stuck.

These changes standardize the agent workflow so every agent -- regardless of platform (Codex, Claude Code, Copilot, Devin) -- follows the same process.

## Changes

| File | Change |
|---|---|
| `AGENTS.md` | Added Agent Workflow section (steps 1-7: branch setup, read context, understand issue, implement, test, open PR, stuck protocol) |
| `.github/PULL_REQUEST_TEMPLATE.md` | New -- structured PR template with clean room compliance checkbox |
| `.github/ISSUE_TEMPLATE/bug.md` | New -- bug report template with repro steps, environment, acceptance criteria |
| `.github/ISSUE_TEMPLATE/feature.md` | New -- feature request template with motivation, affected files, acceptance criteria |

## Key Design Decisions

- **`agent/` branch prefix** -- makes agent branches instantly distinguishable from human work
- **2-attempt stuck limit** -- prevents agents from burning tokens looping on broken approaches
- **Clean room checkbox in PR template** -- forces conscious confirmation on every PR
- **Acceptance criteria required in issue templates** -- single biggest quality lever for agent output

## Testing

- Documentation-only change, no code affected
- Verified all markdown renders correctly

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: AGENTS.md [hunk 1]
```diff
@@ -14,6 +16,98 @@ This is a **clean room reimplementation**. These rules are absolute:
...
+- **Affected files/modules** -- which source and test files you will touch
+- **Edge cases** -- anything mentioned or implied in the issue
+- **Clean room implications** -- does this touch protocol or game logic? If so, the answer must come from `docs/`, not guesswork
+
+Post your plan as a comment on the issue before starting implementation.
+
+### 4. Implement
+
+- Follow existing code style and conventions (see below).
+- Keep changes scoped tightly to the issue. Do not refactor unrelated code, add unrelated features, or "improve" things outside scope.
```
[the remark this answers is not in the document]
Comment (Cadacious) [reply AGENTS.md]: see something, say something. 2020+6 edition.

File: AGENTS.md [hunk 2]
```diff
@@ -14,6 +16,98 @@ This is a **clean room reimplementation**. These rules are absolute:
...
+
+The build must produce **zero warnings** (`-Wall -Wextra -Wpedantic`).
+
+If tests fail:
+- **Your changes caused it** -- fix it before proceeding.
+- **Pre-existing failure unrelated to your changes** -- note it in the PR description. Do not attempt to fix unrelated broken tests.
+
+### 6. Open a Pull Request
+
+Push your branch and open a PR against `main`:
```
Comment (Cadacious) [inline AGENTS.md]: always take in changes from main before opening PRs. this is to ensure we are opening PRs without merge conflicts.

we have a fast moving project and we should always verify something else didnt merge before we open our PR.

whatever happens after we open it happens.

File: AGENTS.md [hunk 3]
```diff
@@ -14,6 +16,98 @@ This is a **clean room reimplementation**. These rules are absolute:
...
+
+```bash
+git push -u origin agent/<issue-number>-<short-slug>
+```
+
+The PR will auto-populate from the repository's pull request template. Fill in every section:
```
Comment (Cadacious) [inline AGENTS.md]: use a more specific template if available liek bug or feature (or anything else wee come up with later)

File: AGENTS.md [hunk 4]
```diff
@@ -14,6 +16,98 @@ This is a **clean room reimplementation**. These rules are absolute:
...
+
+- **Summary** -- what changed and why (1-3 bullets)
+- **Changes** -- list of modified/added files with brief descriptions
+- **Testing** -- how you verified it works (test names, manual steps)
+- **Issue reference** -- `Closes #<number>`
+
+Request review from **@Cadacious**.
+
+### 7. When You Are Stuck
```
Comment (Cadacious) [inline AGENTS.md]: asking for help from the reverse engineering room is acceptable. open an issue with a prompt in the OpenBC repo that can be sent by a meatbag to the claude in the reverse engineering room. that claude will do any needed analysis, write reverse engineering docs, and synthesize it into a clean room acceptable behavior document for our implementation.

File: AGENTS.md [hunk 5]
```diff
@@ -14,6 +16,98 @@ This is a **clean room reimplementation**. These rules are absolute:
...
+
+```bash
+make all && make test
+```
+
+The build must produce **zero warnings** (`-Wall -Wextra -Wpedantic`).
+
+If tests fail:
+- **Your changes caused it** -- fix it before proceeding.
+- **Pre-existing failure unrelated to your changes** -- note it in the PR description. Do not attempt to fix unrelated broken tests.
```
[the remark this answers is not in the document]
Comment (Cadacious) [reply AGENTS.md]: download city awaits the AI agent that does this next.

File: AGENTS.md [hunk 6]
```diff
@@ -1,8 +1,10 @@
...
 
-This file is for OpenAI Codex and other AI coding agents. The canonical project instructions live in **CLAUDE.md** at the repository root. Read that file in full before doing any work -- it contains the architecture, protocol reference, build system, verified facts, and implementation gotchas that apply to all contributors (human or AI).
+This file is for AI coding agents (Codex, Claude Code, Copilot, Devin, etc.). The canonical project instructions live in **CLAUDE.md** at the repository root. Read that file in full before doing any work -- it contains the architecture, protocol reference, build system, verified facts, and implementation gotchas that apply to all contributors (human or AI).
 
-Everything below supplements CLAUDE.md with operational notes for agents that don't auto-load it.
+Everything below supplements CLAUDE.md with operational workflow and conventions.
+
+---
 
 ## Critical: Clean Room Rules
```
[the remark this answers is not in the document]
Comment (aurablacklight) [reply AGENTS.md]: Added -- `SandboxServers/STBC-Reverse-Engineering` is now explicitly called out as strictly off-limits in the clean room rules.

File: AGENTS.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (aurablacklight) [reply AGENTS.md]: Added a "see something, say something" bullet in step 4: agents file new issues for out-of-scope problems rather than fixing them inline.

File: AGENTS.md
[the remark this answers is not in the document]
Comment (aurablacklight) [reply AGENTS.md]: Updated -- build section now shows native `make all` (auto-detects Linux/macOS) and `make PLATFORM=Windows` for cross-compile, with a platform/toolchain table.

[the remark this answers is not in the document]
Comment (aurablacklight) [reply AGENTS.md]: Added -- key conventions now reads: "**Data formats**: JSON for machine-generated/scraped data (ship registries, manifests). TOML for human-managed configuration."
